src/app.tsx

```diff
@@ -1,26 +1,70 @@
-import { useEffect, useSyncExternalStore } from "react";
-import ReactFlow, { Controls, useNodesState } from "reactflow";
+/* eslint-disable react/react-in-jsx-scope */
+import { useCallback, useEffect, useSyncExternalStore } from "react";
+import ReactFlow, {
+	Controls,
+	useNodesState,
+	useReactFlow,
+	useStoreApi,
+	useViewport,
+	type NodeMouseHandler,
+} from "reactflow";
 import "reactflow/dist/style.css";
 import { Postcard } from "./components/postcard";
 import { Title } from "./components/title";
 import { dbStore } from "./store/store";
 import { InfoButton } from "./components/info-dialog/info-button";
 import { InfoDialog } from "./components/info-dialog/info-dialog";
+import { useResetView } from "./components/hooks/use-reset-view";
 
 const nodeTypes = { postcard: Postcard, title: Title };
 
 export default function App() {
+	const isNotMobile = window.innerWidth > 500;
+
 	const ideaNodes = useSyncExternalStore(
 		dbStore.subscribe,
 		dbStore.getSnapshot,
 	);
+
+	const truncatedIdeaNodes = ideaNodes.slice(ideaNodes.length - 30, -1);
+
 	const [nodes, setNodes, onNodesChange] = useNodesState([]);
 
+	const { addSelectedNodes } = useStoreApi().getState();
+
+	const { fitView, getNode } = useReactFlow();
+	const { zoom } = useViewport();
+
 	useEffect(() => {
-		setNodes(ideaNodes);
+		const listener = (event: Event) => {
+			const newNodeID = (event as CustomEvent).detail.id;
+			const newNode = getNode(newNodeID);
+
+			if (newNode) {
+				fitView({ nodes: [newNode], duration: 1200, maxZoom: 1 });
+			}
+			addSelectedNodes([newNodeID]);
+		};
+
+		document.addEventListener("new-idea", listener);
+		document.removeEventListener("new-idea", listener);
+
+		if (isNotMobile) {
+			setNodes(ideaNodes);
+			return;
+		}
+
+		setNodes(truncatedIdeaNodes);
 	}, [ideaNodes]);
 
-	const isNotMobile = window.innerWidth > 500;
+	const handleNodeClick = useCallback<NodeMouseHandler>(
+		(_, node) => {
+			fitView({ nodes: [node], duration: 1200 });
```

Comment:
I removed `maxZoom: 1`. So now the card is zoomed in to the maxZoom level defined in line 79. Now we don't need the "avoid zooming out when clicking on card" functionality anymore since, we already are at the maxZoom level.